ffw/NavigationRPC.js

```diff
@@ -326,18 +326,33 @@ FFW.Navigation = FFW.RPCObserver.create(
                 webmSupport: can_play
               };
 
-              const preffered = SDL.NavigationController.stringifyCapabilityItem({
-                preferredResolution: {
-                  resolutionWidth:  request.params.config.width,
-                  resolutionHeight: request.params.config.height
+              const set_preferred = function(width, height, scale) {
+                const preferred = SDL.NavigationController.stringifyCapabilityItem({
+                  preferredResolution: {
+                    resolutionWidth:  width,
+                    resolutionHeight: height
+                  },
+                  scale: scale
+                });
+                const preset_list = SDL.NavigationController.getVideoStreamingCapabilitiesList();
+                const index = preset_list.indexOf(preferred);
+                Em.Logger.log("App Preferred Index: " + index)
+                if (index >= 0 && index !== app_model.resolutionIndex) {
+                  Em.Logger.log(`Switching video streaming preset to: ${preferred}`);
+                  SDL.NavigationController.model.set('resolutionIndex', index);
```

Comment:
@JackLivio issue is still seen with that fix. Looks like you are trying to set resolution index in a generic data model, but not for a specific application model. It won't work because resolution index property has been moved to application model. You should change it to:
```suggestion
app_model.set('resolutionIndex', index);
```

Reply:
@JackLivio with the suggested changes it works as expected on my side

Reply:
@AKalinich-Luxoft Thank you for the suggestion! Comitted